sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/ExecutorServiceParallelExecutor.java

```diff
@@ -142,7 +159,7 @@ public void scheduleConsumption(
     if (bundle != null && isKeyed(bundle.getPCollection())) {
       final StepAndKey stepAndKey =
           StepAndKey.of(transform, bundle == null ? null : bundle.getKey());
-      transformExecutor = getSerialExecutorService(stepAndKey);
+      transformExecutor = executorServices.getUnchecked(stepAndKey);
```

Comment:
I would add a comment that the lifetime of the callable since its scheduled with the  serial/parallel executor service keeps the transform executor in scope until the callable is finished.